processes/dags/pubmed_migration.py

```diff
@@ -111,59 +105,247 @@ def make_tarfile(output_filepath: str, source_dir: str) -> None:
-        delete_file(filepath_tar_gz)
-        make_tarfile(filepath_tar_gz, o_path)
-        upload_file_to_S3_with_hook(
-            filepath_tar_gz, filename_tar_gz, 'supreme-acrobat-data')
+        create_dir(json_dir)
+
+        basename = os.path.basename(key)
+        local_path = os.path.join(temp_dir, basename)
+        o_path = extract_original_name(local_path)
+        o_path_basename = os.path.basename(o_path)
+
+        obj = s3_hook.get_key(key, src_bucket_name)
+        obj.download_file(local_path)
+
+        extract_file(local_path, temp_dir)
+
+        glob_path = os.path.join(o_path, "*", "*.nxml")
+
+        filenames = [f for f in glob.glob(glob_path)]
+        logging.info("filenames = %s" % filenames)
+        json_list = []
+
+        for filename in filenames[0:4]:
+            fname, ext = os.path.splitext(os.path.basename(filename))
+            json_filename = ".".join((fname, "json"))
+            jl_filename = ".".join((fname, "jsonl"))
+            json_path = os.path.join(json_dir, json_filename)
+            jl_path = os.path.join(json_dir, jl_filename)
+            new_json = export_case_report_json(filename)
+            json_list.append(new_json)
+
+        with open(jl_path, 'w') as outfile:
+            for entry in json_list:
+                json.dump(entry, outfile)
+                outfile.write('\n')
+
+        key = os.path.join(dest_path, os.path.basename(o_path_basename + ".jsonl"))
+        s3_hook.load_file(jl_path, key, bucket_name=dest_bucket_name, replace=True)
+
+        jsons = [f for f in glob.glob(os.path.join(json_dir, "*.json"))]
+        logging.info("json list = %s" % str(len(json_list)))
         delete_dir(temp_dir)
 
-        try:
-            ftp.quit()
-        except:
-            None
+
+def transform_pubmed_data_failure_callback(context) -> None:
+    delete_temp()
 
 
-def extract_pubmed_data_failure_callback() -> None:
+def update_mongo() -> None:
+    """Updates MongoDB caseReports
+    """
     root_dir = '/usr/local/airflow'
-    temp_dir = root_dir + '/' + 'temp'
-    delete_dir(temp_dir)
+    temp_dir = os.path.join(root_dir, 'temp')
+    json_dir = os.path.join(temp_dir, 'json')
+    src_path = 'case_reports/pubmed/json/'
+    src_bucket_name = 'supreme-acrobat-data'
+    wildcard_key = 'case_reports/pubmed/json/*.*'
+    klist = s3_hook.list_keys(
+        src_bucket_name, prefix=src_path, delimiter='/')
+    key_matches = []
+    key_matches = [k for k in klist if fnmatch.fnmatch(k, wildcard_key)]
+
+    logging.info(key_matches)
+
+    for key in key_matches:
+        basename = os.path.basename(key)
+        local_path = os.path.join(temp_dir, basename)
+        o_path = extract_original_name(local_path)
+        docs = []
+        results = []
+
+        obj = s3_hook.get_key(key, src_bucket_name)
+        file_content = obj.get()['Body'].read().decode('utf-8')
+        results = file_content.splitlines()
+        for result in results:
```

Comment:
Nothing wrong here, but can you optimize this by not storing results?
You can do something like this:
```for line in file_content.splitlines():```